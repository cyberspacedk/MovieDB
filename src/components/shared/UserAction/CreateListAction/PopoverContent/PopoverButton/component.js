import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'antd';

const PopoverButton = ({ item, handlerAddMovieToList }) => (
  <Button type="link" onClick={handlerAddMovieToList}>
    {item.name}
  </Button>
);

PopoverButton.propTypes = {
  item: PropTypes.arrayOf(PropTypes.object),
  handlerAddMovieToList: PropTypes.func,
};

export default PopoverButton;
