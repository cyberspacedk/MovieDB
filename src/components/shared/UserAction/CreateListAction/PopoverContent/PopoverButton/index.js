import PopoverButtonContainer from './container';

export default PopoverButtonContainer;
