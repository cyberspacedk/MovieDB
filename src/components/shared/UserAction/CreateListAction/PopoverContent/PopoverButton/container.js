import React from 'react';
import PropTypes from 'prop-types';
import PopoverButton from './component';

const PopoverButtonContainer = ({ movieId, addMovieToList, item }) => {
  const handlerAddMovieToList = () => {
    addMovieToList(item.id, movieId);
    console.log(item.id, movieId);
  };
  return (
    <PopoverButton item={item} handlerAddMovieToList={handlerAddMovieToList} />
  );
};

PopoverButtonContainer.propTypes = {
  movieId: PropTypes.number,
  item: PropTypes.arrayOf(PropTypes.object),
  addMovieToList: PropTypes.func,
};
export default PopoverButtonContainer;
