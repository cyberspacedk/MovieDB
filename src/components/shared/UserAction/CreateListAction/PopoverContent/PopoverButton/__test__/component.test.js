import React from 'react';
import { shallow } from 'enzyme';
import PopoverButton from '../component';

describe('Component: PopoverButton', () => {
  const props = {
    handlerAddMovieToList: jest.fn(),
    item: { name: 'John' },
  };

  const wrapper = shallow(<PopoverButton {...props} />);

  it('Match its snapshot.', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
