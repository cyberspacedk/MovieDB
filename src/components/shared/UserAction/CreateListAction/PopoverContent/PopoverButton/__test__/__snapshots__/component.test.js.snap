// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PopoverButton Match its snapshot. 1`] = `
<Button
  block={false}
  ghost={false}
  htmlType="button"
  loading={false}
  onClick={[MockFunction]}
  type="link"
>
  John
</Button>
`;
